tests/test_config.py

```diff
@@ -435,17 +446,37 @@ def test_merge():
         else:
             assert config2[key] == valid_params[key][2]
 
-    # If conflicts, kwargs takes precedence
+    # If conflicts, kwargs takes precedence ...
     kwargs['ra_col'] = 'alpha2000'
     config2 = treecorr.config.merge_config(config1, kwargs, treecorr.Catalog._valid_params)
     assert config2['ra_col'] == 'alpha2000'
 
+    # ... except when the value in kwargs is None.
+    kwargs = {'bin_size': 0.06, 'nbins': None}
+    config2 = treecorr.config.merge_config(config1, kwargs, treecorr.BinnedCorr2._valid_params)
+    assert config2['bin_size'] == 0.06
+    assert config2['nbins'] == config1['nbins']
+    assert kwargs['nbins'] == config2['nbins']
+    assert kwargs['nbins'] is not None
+    # Note: config2 is not in a valid state to be used to initialize BinnedCorr2.
```

Comment:
I'd like your opinions on this behavior, Mike. Essentially, one cannot unset a config by setting it as None explicitly in kwargs. If this is not a desired behavior, I'm happy to make changes to not make an exception for `None`.

Reply:
Hm.  Good point.  This should do whatever is least surprising to the user.  And I think if a user put `config=config, bin_size=0.06, nbins=None` explicitly in the kwargs, they would expect it to override whatever `nbins` was in the config.  So probably we should switch that to work the other way.
```
    assert config2['nbins'] is None
```